vita3k/io/src/io.cpp

```diff
@@ -482,13 +490,15 @@ SceUID read_dir(IOState &io, const SceUID fd, emu::SceIoDirent *dent, const std:
             if (fs::is_regular_file(cur_path)) {
                 if (stat_file(io, file_path.c_str(), &dent->d_stat, pref_path, base_tick, export_name) < 0)
                     return IO_ERROR(SCE_ERROR_ERRNO_EMFILE);
+                else
+                    return 1; // move to the next file
             }
             if (fs::is_directory(cur_path)) {
-                const auto dir_fd = open_dir(io, file_path.c_str(), pref_path, export_name);
-                return read_dir(io, dir_fd, dent, pref_path, base_tick, export_name);
+                stat_file(io, file_path.c_str(), &dent->d_stat, pref_path, base_tick, export_name);
+                return 1;
```

Comment:
not sure what `return 1;` means, the func's supposed to return a `SceUID` isn't it?
(not your fault since it's existing code but it'd be nice to explain or clean it up)

Reply:
`SceIoDread` returns an int.
this is what vitasdk says:
```
= 0 - No more directory entries left
> 0 - More directory entries to go
< 0 - Error
```

Reply:
Ok, would be cool to have an enum for this.